Con 2128 setup new repos

Carsten89 opened the pull request.

Carsten89 committed:
initial commit

Carsten89 committed:
initial commit

Carsten89 committed:
fixing tests

Carsten89 committed:
fixing tests

Carsten89 committed:
fixing tests

Carsten89 committed:
fixing tests

Carsten89 committed:
fixing tests

Carsten89 committed:
codestyle

Carsten89 committed:
Merge branch 'master' into CON-2128-setup-new-repos

# Conflicts:
#	.babelrc

fkloes reviewed (commented):
Please update the LICENSE.md. Currently it contains placeholders for the year and the copyright owner.

fkloes commented on plugins/Base.js:
```diff
@@ -0,0 +1,134 @@
+   * @param {string} trackerName The name of the tracker that is represented by the plugin
+   * @param {Object} options Configuration for the plugin
+   * @param {Object} extendedDefaults Additional default options that are
+   *                                  needed by the inherited class
```
Please check the indentation

fkloes replied:
changed

Carsten89 replied:
done

fkloes commented on plugins/Base.js:
```diff
@@ -0,0 +1,134 @@
+      /**
+       * Function to register a plugin for the event
+       *
+       * @param {Function} callback  Function that is called if the event occurs
```
Please check the indentation for the complete JSDoc

fkloes replied:
changed

Carsten89 replied:
done

fkloes commented on plugins/Base.js:
```diff
@@ -0,0 +1,134 @@
+  /**
+   * Helper function to register a plugin for a specific event. Can be overwritten in the plugins.
+   *
+   * @param {string}   eventName Name of the event
```
Please check the indentation for the complete JSDoc

fkloes replied:
changed

Carsten89 replied:
done

fkloes commented on plugins/trackers/FbPixel.js:
```diff
@@ -0,0 +1,236 @@
+   */
+  initPlugin() {
+    // Load the fb pixel tracking sdk
+    /*eslint-disable */
```
This is treated as an error and might cause a rejected commit, if we introduce commit hooks. A solution - which causes some warnings - is to wrap this block into a /* eslint-disable eslint-comments/no-unlimited-disable */ block

fkloes replied:
Please also insert a space after /* - same at /*eslint-enable */

fkloes replied:
changed

fkloes commented on plugins/trackers/FbPixel.js:
```diff
@@ -0,0 +1,236 @@
+  /**
+   * Constructor
+   *
+   * @param {Object}  options                   Common Tracking Configuration
```
Please check the indentation for the complete JSDoc

fkloes replied:
changed

Carsten89 replied:
done

fkloes commented on plugins/trackers/FbPixel.js:
```diff
@@ -0,0 +1,236 @@
+
+    /*
+     * The pixel tracking have a PageView und ViewContent Event. The PageView event should be called
+     * On every page. It has no params
```
Please correct this comment - `On` with a small character and a `.` at the end of it.

fkloes replied:
changed

fkloes commented on plugins/trackers/FbPixel.js:
```diff
@@ -0,0 +1,236 @@
+   * @param {Object} data Converted data from the parent plugin
+   * @returns {Object} Params for the fb event
+   */
+  getParamsForAddtoCartAndWishlist(data) {
```
Please correct the name of the function ... AddToCart should have an upper case t

fkloes replied:
done

fkloes commented on plugins/trackers/GaBase.js:
```diff
@@ -0,0 +1,303 @@
+  }
+
+  /**
+   * Creates an ga classic child tracker
```
Typo `a` instead of `an`

fkloes replied:
done

fkloes commented on plugins/trackers/GaClassic.js:
```diff
@@ -0,0 +1,146 @@
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/* global _gaq */
```
Maybe add a blank line after this line to improve readability

fkloes replied:
done

fkloes commented on plugins/trackers/GaClassic.js:
```diff
@@ -0,0 +1,146 @@
+   */
+  initPlugin() {
+    // Load the SDK
+    /* eslint-disable */
```
This will cause commit rejections when we have commit hooks in the project

fkloes replied:
done

fkloes commented on plugins/trackers/GaClassic.js:
```diff
@@ -0,0 +1,146 @@
+   */
+  initPlugin() {
+    // Load the SDK
+    /* eslint-disable */
```
This will cause commit rejections when we have commit hooks in the project

fkloes replied:
done

fkloes commented on plugins/trackers/GaUniversal.js:
```diff
@@ -0,0 +1,200 @@
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/* global ga */
```
Please add a blank line afterwards to improve readability

fkloes replied:
done

fkloes commented on plugins/trackers/GaUniversal.js:
```diff
@@ -0,0 +1,200 @@
+// SgData access type constant for apps
+const ACCESS_TYPE_APP = 'App';
+
+/*
```
Please add an additional `*`

fkloes replied:
done

fkloes commented on plugins/trackers/GaUniversal.js:
```diff
@@ -0,0 +1,200 @@
+
+/*
+ * Dimensions and metrics
+ * See: https://shopgate.atlassian.net/wiki/display/CONSUMER/Google+Analytics
```
This link doesn't work anymore since we moved to our new Jira. If this information is still relevant, we need to move it somewhere else.

fkloes commented on plugins/trackers/GaUniversal.js:
```diff
@@ -0,0 +1,200 @@
+  /**
+   * Constructor
+   *
+   * @param {Object}  options                   Common Tracking Configuration
```
Please check the indentation for the complete JSDoc

fkloes replied:
done

fkloes commented on plugins/trackers/GaUniversal.js:
```diff
@@ -0,0 +1,200 @@
+  initPlugin() {
+    // Load the SDK (if the code is executed form a unit test, always load the sdk)
+    if (typeof window.ga !== 'function' || (typeof global !== 'undefined' && typeof global.it === 'function')) {
+      /* eslint-disable */
```
This is treated as an eslint error and will cause rejections, if the have our commit hooks.

fkloes replied:
done

fkloes commented on plugins/trackers/GaUniversal.js:
```diff
@@ -0,0 +1,200 @@
+    // Send custom variables to sdk
+    ga('shopgate.set', DIMENSION_SHOP_NUMBER, window.sgData.shop.shop_number.toString());
+
+    if (window.sgData.device.access === ACCESS_TYPE_APP) {
```
We should check if there is a better solution this this

Carsten89 replied:
maybe later...

fkloes reviewed (changes requested):
- The mock data for the tests cause linter errors. Commits will be probably rejected, if we have commit hooks.
- The license comments of all files have a wrong format. A block comment starts with `/**`, not `/*`

fkloes commented on core/AppHandler.js:
```diff
@@ -0,0 +1,180 @@
+import { SGAction } from 'tracking-helper';
+
+/**
+ * Handler for the communication between the tracking plugins and the app (LibShopgate_App)
```
We don't have a `LibShopgate_App` in the new system

fkloes replied:
Removed

fkloes commented on core/Core.js:
```diff
@@ -0,0 +1,401 @@
+  /**
+   * Notify plugins helper
+   *
+   * @param {Object} rawData   Object with tracking data for the event
```
Please check the indentation for the complete JSDoc

fkloes replied:
Changed

fkloes commented on core/Core.js:
```diff
@@ -0,0 +1,401 @@
+   * @returns {void}
+   */
+  notifyHelper(rawData, eventName, page, scope = {}) {
+    // Exit if the user opt out. but not for the explicit add or remove tracker events
```
Typo... big `B` at the beginning of the new sentence

fkloes replied:
changed

fkloes commented on core/Core.js:
```diff
@@ -0,0 +1,401 @@
+
+    console.info('[TRACK]', eventName, rawData);
+
+    /*
```
Please add a 2nd `*` to have a valid block comment

fkloes replied:
changed

fkloes commented on core/Core.js:
```diff
@@ -0,0 +1,401 @@
+  /**
+   * Notify plugins
+   *
+   * @param {Object} rawData   Object with tracking data for the event
```
Please check the indentation for the complete JSDoc

fkloes replied:
changed

fkloes commented on core/Core.js:
```diff
@@ -0,0 +1,401 @@
+      // No classic sdk
+
+      if (formElement) {
+        /*
```
Please add a 2nd `*`

fkloes replied:
changed

fkloes commented on helpers/formatHelpers.js:
```diff
@@ -0,0 +1,388 @@
+ * Gets the value at path of object. If the resolved value is undefined,
+ * the defaultValue is used in its place.
+ *
+ * @param   {Object} object         The object to query
```
Please check the indentation for the complete JSDoc

fkloes replied:
done

fkloes commented on helpers/formatHelpers.js:
```diff
@@ -0,0 +1,388 @@
+  /**
+   * Recursive callable function to traverse through a complex object
+   *
+   * @param   {Object} currentData             The current data that shall be investigated
```
Please check the indentation for the complete JSDoc

fkloes replied:
done

fkloes commented on helpers/formatHelpers.js:
```diff
@@ -0,0 +1,388 @@
+    const currentPathSegment = pathSegments[currentPathSegmentIndex];
+    const nextPathSegmentIndex = currentPathSegmentIndex + 1;
+
+    /*
```
Please convert to a proper block comment

fkloes replied:
done

fkloes commented on helpers/formatHelpers.js:
```diff
@@ -0,0 +1,388 @@
+/**
+ * Converts a numeric value into a suitable one for the unified tracking data
+ *
+ * @param   {*}                numericValue The value that shall be converted
```
Please check the indentation for the complete JSDoc

fkloes replied:
done

fkloes commented on helpers/formatHelpers.js:
```diff
@@ -0,0 +1,388 @@
+  }
+
+  /*
+   Setup the RegExp. It matches leading and trailing occurrences
```
Please convert to a proper block comment

fkloes replied:
done

fkloes commented on helpers/formatHelpers.js:
```diff
@@ -0,0 +1,388 @@
+ * @param {Array} products Items from sgData
+ * @returns {UnifiedAddToWishlistItem[]} Data for the unified items
+ */
+function formatFavouriteListItems(products) {
```
I would recommend to rename this function and adjust the comment. The function is used within the `addToWishlist` `dataFormatHelper`

fkloes commented on helpers/formatHelpers.js:
```diff
@@ -0,0 +1,388 @@
+}
+
+/*
+ Storage for helper functions that transforms raw data
```
Please convert to a proper block comment

fkloes replied:
done

fkloes commented on helpers/formatHelpers.js:
```diff
@@ -0,0 +1,388 @@
+  let link = get(rawData, 'page.link');
+
+  if (link.indexOf('sg_app_resources') !== -1) {
+    /*
```
Please convert to a proper block comment

fkloes replied:
done

fkloes commented on helpers/formatHelpers.js:
```diff
@@ -0,0 +1,388 @@
+  link = new SGLink(link);
+
+  /*
+   In splittedPath we have the action as the first array entry
```
Please convert to a proper block comment

fkloes replied:
done

fkloes commented on helpers/formatHelpers.js:
```diff
@@ -0,0 +1,388 @@
+  splittedPath.shift();
+
+  /*
+   All pages have action as type and a parsed page title without shop name as name
```
Please convert to a proper block comment

fkloes replied:
done

fkloes commented on helpers/formatHelpers.js:
```diff
@@ -0,0 +1,388 @@
+    get(rawData, 'page.name');
+
+  /*
+   Category, product and product related pages should have productnumber/categorynumber as id
```
Please convert to a proper block comment

fkloes replied:
done

fkloes commented on helpers/formatHelpers.js:
```diff
@@ -0,0 +1,388 @@
+  };
+};
+
+dataFormatHelpers.setCampaignWithUrl = rawData => ({
```
Please add a comment

Carsten89 replied:
done

fkloes commented on helpers/helper.js:
```diff
@@ -0,0 +1,274 @@
+import logger from '@shopgate/pwa-core/classes/Logger';
+import { hex2bin } from '@shopgate/pwa-common/helpers/data';
+
+/*
```
Please convert to a proper block comment

fkloes replied:
done

fkloes commented on helpers/helper.js:
```diff
@@ -0,0 +1,274 @@
+
+  /**
+   * Encode the url and takes care of double encoding
+   * @param {string} string - string to be encoded
```
Please correct the comment

fkloes replied:
done

fkloes commented on helpers/helper.js:
```diff
@@ -0,0 +1,274 @@
+  /**
+   * Parses url and extracts path, query, action, splittedPath, ...
+   *
+   * @param {string} incomingUrl urlToSanitize to parse
```
Param doc looks strange

fkloes replied:
changed

fkloes commented:
The alias for the `tracking-helper` doesn't seem to be a good solution. Wouldn't it be better to define a "proxy module" which exports either the content of the new helpers or the content of the legacy helpers - depending on an environment variable which is set within the `webpack.config`? 

By default it could export the content of the new file, so that we only have to set this variable within the `webpack.config` of the legacy system.

fkloes committed:
Implemented feedback

- importing the tracking helper directly from the helpers folder
- corrected comment formatting

Carsten89 committed:
implemented feedback

Carsten89 committed:
Merge branch 'CON-2128-setup-new-repos' of https://github.com/shopgate/tracking-core into CON-2128-setup-new-repos

Carsten89 committed:
implemented feedback
